Experimental SSS

dbkr opened the pull request:
Working branch to get SSS functional on element-web.

Requires https://github.com/matrix-org/matrix-js-sdk/pull/4400

Current status is that it syncs, but the room lists don't load until the next sync request, so take 30s to appear. This is because element web doesn't add the requests for the lists until after sliding sync has started, and SSS doesn't support txn_id which the SS impl uses to track requests for cancellation (it looks like previously it would have started the second request and the shortly after, canceled it and made a new one with the room lists).

<!-- Thanks for submitting a PR! Please ensure the following requirements are met in order for us to review your PR -->

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/matrix-react-sdk)

dbkr committed:
Experimental SSS

Working branch to get SSS functional on element-web.

Requires https://github.com/matrix-org/matrix-js-sdk/pull/4400

kegsay committed:
Adjust tests to use new behaviour

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/element-hq/matrix-react-sdk?pullRequest=36) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/element-hq/matrix-react-sdk?pullRequest=36) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/element-hq/matrix-react-sdk?pullRequest=36) it.</sub>

kegsay committed:
Remove well-known proxy URL lookup; always use native

This is actually required for SSS because otherwise it would use
the proxy over native support.

kegsay committed:
Linting

kegsay committed:
Merge branch 'develop' into dbkr/sss

kegsay committed:
Debug logging

kegsay committed:
Control the race condition when swapping between rooms

kegsay committed:
Dont' filter by space as synapse doesn't support it

kegsay committed:
Remove SS code related to registering lists and managing ranges

- Update the spidering code to spider all the relevant lists.
- Add canonical alias to the required_state to allow room name calcs to work.

Room sort order is busted because we don't yet look at `bump_stamp`.

kegsay committed:
User bumpStamp if it is present

kegsay committed:
Drop initial room load from 20 per list to 10

kegsay committed:
Half the batch size to trickle more quickly

t3chguy commented:
We have made the decision to merge matrix-react-sdk into https://github.com/element-hq/element-web - if you are still interested in this contribution please move it to that repository. Closing so it does not remain as a zombie
